Caltech-IPAC/firefly
# DM-12971: various HiPS fixes and improvements

Big Pull request. Mostly a log of little change and factoring.  _See reviewer notes at bottom._

Various HiPS fixes and improvements and a little Java refactoring.

   - make all text draw with canvas
   - Removed div text drawing.  Now all text drawing uses canvas
   - make restore button work with HiPS
   - fixed - color dialog now aware of HiPS
   - firefly server can act as hips proxy - hips images are brought to firefly server first
   - cached HiPS images check for modified
   - hips cache at 100 images
   - added hips with remote viewer Api
   - added hips JS api
   - give more level of hips zoom
   - code cleanup: removed ClientLog from all java
   - hide thumbnail better for HiPS
   - use `obs_title` hips property
   - added direct image download without using server
   - added upload of image and put to workspace
   - a little clean up of AnyFileUpload (it needs more)

_To Test:_

  - Make sure text show up as expected on the drawing layers
  - Hit the restore button while viewing hips
  - change server work area `workarea/firefly/HiPS` to make sure cached hips images appear
  - bring up `http://localhost:8080/firefly/demo/ffapi-highlevel-test.html and http://localhost:8080/firefly/demo/ffapi-slate-test2.html and check for HIPS
  - use download panel to download a image or hips to local disk and workspace,  (for workspace you must edit `firefly/src/firefly/config/app.prop` set `ignore.auth=false`  )

_For Reviewer:_

 - **Java**: Only review `AnyFileDownload.java` and `AnyFileUpload.java`. All other java changes are logger refactoring  and can be ignored.
 - **JavaScript**: 
       - The DrawObj type classes and `Drawer.js` are just a refactor to change it to canvas based drawing. Just review `DrawUtil.drawTextCanvas`
       -  review: `FitsDownloadDialog.jsx`, `WebUtil.jsx`, `PlotViewUtil.js`, `ApiHighlevelBuild.js`, `ApiViewer.js` 
 - **html** - very small changes that don't need review
